pkg/runner/runner.go

```diff
@@ -73,8 +74,10 @@ func ParseFlags(cfg *Config) (*Config, error) {
 	flag.StringVar(&migrateOption, "migrate", "true", "Update the Prometheus SQL to the latest version. Valid options are: [true, false, only]")
 	flag.BoolVar(&cfg.UseVersionLease, "use-schema-version-lease", true, "Prevent race conditions during migration")
 	flag.BoolVar(&cfg.InstallTimescaleDB, "install-timescaledb", true, "Install or update the TimescaleDB extension")
+	flag.BoolVar(&cfg.ReadOnly, "read-only", false, "Read-only mode. Don't write to database. Useful when pointing adapter to read replica")
 	envy.Parse("TS_PROM")
-	flag.Parse()
+	// Ignore errors; CommandLine is set for ExitOnError.
+	_ = flag.CommandLine.Parse(args)
```

Comment:
Not sure, but I think error is returned here if there is any parsing issue or a `--help` is passed. In both the cases (especially for later), we should stop or just return here.

Reply:
`ExitOnError` means the program will call `os.Exit` on an error when parsing flags. This is the default behavior, this line change does the same thing as `flag.Parse`, only takes in arguments manually instead of using `os.Args`.

https://golang.org/pkg/flag/#ExitOnError